### PR TITLE
Improve backtrace catching on server failures in CI for stress tests

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow-up for: #19524
Cc: @alexey-milovidov

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/pull/19524 does not handle SIGABRT deliberately.
Because there should not be the cases when we cannot print our own diagnostics.

**Comment by alexey-milovidov** [comment]
I see the motivation here: https://github.com/ClickHouse/ClickHouse/pull/19580#issuecomment-778333022, let's try...

File: docker/test/stress/run.sh [hunk 1]
```diff
@@ -8,6 +8,9 @@ dpkg -i package_folder/clickhouse-server_*.deb
 dpkg -i package_folder/clickhouse-client_*.deb
 dpkg -i package_folder/clickhouse-test_*.deb
 
+# allow ptrace from other users
+sudo sysctl kernel.yama.ptrace_scope=0
```
Comment (alexey-milovidov) [inline docker/test/stress/run.sh:12]: I hear about it the first time. Is it required in fuzzer as well?

Comment (azat) [reply docker/test/stress/run.sh:12]: fuzzer does not need this because server does not run as a service, so the user matches and no sudo is required

Comment (azat) [reply docker/test/stress/run.sh:12]: So the problem is not in yama (LSM), but the containers I guess.
So let's just not start clickhouse-server as a service and start as a regular process instead, then this should work

Comment (alexey-milovidov) [reply docker/test/stress/run.sh:12]: Ok.

**Comment by azat** [comment]
This is also required in other places (like fast-test - https://clickhouse-test-reports.s3.yandex.net/20477/8ad20fcee5aaa642c2a2dd873d02103692d554f4/fast_test.html#fail1)

**Comment by alexey-milovidov** [comment]
Note that it should not prevent receiving fatal messages with stack trace in client.

**Comment by azat** [comment]
>Note that it should not prevent receiving fatal messages with stack trace in client.

Right. Actually I don't think that this should break this, since the server will continue after gdb will print stacktraces

**Comment by azat** [comment]
Stress tests are okay, but fasttest not:

https://clickhouse-test-reports.s3.yandex.net/20462/014c11af7fcaa60d62c10b1497bbdc02526d180c/fast_test/runlog.out.log

```
/ClickHouse/docker/test/fasttest/run.sh: line 121: gdb: command not found
```

By some reason fasttest image wasn't updated, because that layer was in cache https://clickhouse-test-reports.s3.yandex.net/20462/014c11af7fcaa60d62c10b1497bbdc02526d180c/push_to_dockerhub/taglatest.out.1.log

```
 ---> Using cache
 ---> 9446b5f83627
Step 6/17 : RUN apt-get update     && apt-get install         apt-transport-https         bash         brotli         build-essential         ca-certificates         ccache         clang-${LLVM_VERSION}         clang-tidy-${LLVM_VERSION}         cmake         curl         lsof         expect         fakeroot         git         gdb         gperf         lld-${LLVM_VERSION}         llvm-${LLVM_VERSION}         moreutils         ninja-build         psmisc         python3         python3-pip         python3-lxml         python3-requests         python3-termcolor         rename         software-properties-common         tzdata         unixodbc        --yes --no-install-recommends
 ---> Using cache
```

**Comment by azat** [comment]
72c61da does not helps with avoiding using docker cache for "new" layers, maybe it should be reseted somehow? (@alesapin ?)

**Comment by azat** [comment]
Lack of backtrace in stress tests happens one more time - https://clickhouse-test-reports.s3.yandex.net/20477/18a147b3179e5931d89395d5e46b4d833d4bef05/stress_test_(memory).html#fail1

**Review by alexey-milovidov — APPROVED** [review]

**Comment by azat** [comment]
>72c61da does not helps with avoiding using docker cache for "new" layers, maybe it should be reset somehow? (@alesapin ?)

Let's try install it as a separate layer then - c879719e7c1ed188cdd81a73148575cdbbacaf13

**Comment by azat** [comment]
>Let's try install it as a separate layer then - c879719

Still no luck

**Comment by azat** [comment]
>Marker check Pending — Starting checks (No special conditions apply)

Looks hanged

@alexey-milovidov can you merge this? (maybe it will reproduce the problem and will have full stacktrace, in the meantime I will look into this https://github.com/ClickHouse/ClickHouse/pull/16481#issuecomment-781980807)

**Pull request merged**